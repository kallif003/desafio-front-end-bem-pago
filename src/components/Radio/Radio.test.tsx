import * as React from "react"
import { render } from "@testing-library/react"
import userEvent from "@testing-library/user-event"

import { RadioDiv, Radio, TextRadio, RadioContainer } from "./"

describe("Radio components", () => {
	it("Renders RadioDiv properly", () => {
		const { container } = render(<RadioDiv>Test</RadioDiv>)
		expect(container.firstChild).toMatchSnapshot()
	})
	it("Renders Radio properly", () => {
		const { container } = render(<Radio>Test</Radio>)
		expect(container.firstChild).toMatchSnapshot()
	})
	it("Renders TextRadio properly", () => {
		const { container } = render(<TextRadio>Test</TextRadio>)
		expect(container.firstChild).toMatchSnapshot()
	})
	it("Renders RadioContainer properly", () => {
		const { container } = render(<RadioContainer>Test</RadioContainer>)
		expect(container.firstChild).toMatchSnapshot()
	})
	it("test Radio button ", () => {
		const { container } = render(<Radio>Test</Radio>)
		userEvent.click(container)
	})
})
