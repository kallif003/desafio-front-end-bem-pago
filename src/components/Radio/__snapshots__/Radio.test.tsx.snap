// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio components Renders RadioContainer properly 1`] = `
<div
  class="sc-gsDKAQ fzACFK"
>
  Test
</div>
`;

exports[`Radio components Renders RadioDiv properly 1`] = `
<div
  class="sc-bdvvtL brsFqP"
>
  Test
</div>
`;

exports[`Radio components Renders TextRadio properly 1`] = `
<label
  class="sc-hKwDye XYNJj"
>
  Test
</label>
`;
